### PR TITLE
Fixcoloring

### PR DESCRIPTION
**Pull request by mosuem**
* Makes the coloring handle nested message format cases
* Implements coloring of arguments
* Disallows comments in JSON

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -89,44 +90,88 @@ const pluralDecoration = vscode.window.createTextEditorDecorationType({
...
 
-function parse(value: string, level: number): string | undefined {
-	if (selectRegex.exec(value) !== null) {
-		return 'select';
-	} else if (pluralRegex.exec(value) !== null) {
-		return 'plural';
-	} else {
-		return 'arg';
+function colorPart(value: string, offset: number, colorMap: Map<vscode.TextEditorDecorationType, vscode.Range[]>, editor: vscode.TextEditor, isOuter: boolean) {
+	const vals = XRegExp.matchRecursive(value, '\\{', '\\}', 'g');
```
Comment (DanTup) [inline src/extension.ts:138]: What happens if the braces are unbalanced (eg. because the user is typing)?

Comment (DanTup) [reply src/extension.ts:138]: also: does this format allow for the user to enter literal `{`'s with an escape character? (if so, will this handle them correctly?)

**Review by DanTup — APPROVED** [review]
Few questions, but otherwise LGTM!

It could be worth splitting this into a separate class in another file (since `extension.ts` might get quite large as more features are added), and adding some kind of tests will make it easier to ensure potential edge cases (like unbalanced braces) are handled reasonably.

File: src/extension.ts (hunk 1 above)
Comment (mosuem) [reply src/extension.ts]: Unbalanced brackets are handled gracefully, and now escape characters also work! Good catch.

File: src/extension.ts [hunk 2]
```diff
@@ -87,46 +88,95 @@ const pluralDecoration = vscode.window.createTextEditorDecorationType({
...
-	if (selectRegex.exec(value) !== null) {
-		return 'select';
-	} else if (pluralRegex.exec(value) !== null) {
-		return 'plural';
-	} else {
-		return 'arg';
+function colorPart(value: string, offset: number, colorMap: Map<vscode.TextEditorDecorationType, vscode.Range[]>, editor: vscode.TextEditor, isOuter: boolean) {
+	const vals = XRegExp.matchRecursive(value, '\\{', '\\}', 'g', {
+		escapeChar: '\''
+	});
```
Comment (DanTup) [inline src/extension.ts:141]: Does this need doing on line 162 too?

Comment (mosuem) [reply src/extension.ts:141]: Yes, this is addressed in #6 .
